contrib/IECoreAlembic/src/IECoreAlembic/GeomBaseAlgo.cpp

```diff
@@ -0,0 +1,238 @@
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "IECore/MessageHandler.h"
+
+#include "IECoreAlembic/GeomBaseAlgo.h"
+#include "IECoreAlembic/IGeomParamTraits.h"
+
+using namespace Alembic::Abc;
+using namespace Alembic::AbcGeom;
+using namespace IECore;
+using namespace IECoreAlembic;
+
+//////////////////////////////////////////////////////////////////////////
+// Internal utilities
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+IECore::PrimitiveVariable::Interpolation interpolationFromScope( Alembic::AbcGeom::GeometryScope scope )
+{
+	switch( scope )
+	{
+		case kConstantScope :
+			return PrimitiveVariable::Constant;
+		case kUniformScope :
+			return PrimitiveVariable::Uniform;
+		case kVaryingScope :
+			return PrimitiveVariable::Varying;
+		case kVertexScope :
+			return PrimitiveVariable::Vertex;
+		case kFacevaryingScope :
+			return PrimitiveVariable::FaceVarying;
+		default :
+			return PrimitiveVariable::Invalid;
+	}
+}
+
+
+template<typename DataType, typename GeomParam>
+struct ApplyGeometricInterpretation
+{
+
+	static void apply( DataType *data )
+	{
+	};
+
+};
+
+template<typename T, typename GeomParam>
+struct ApplyGeometricInterpretation<GeometricTypedData<T>, GeomParam>
+{
+
+	static void apply( GeometricTypedData<T> *data )
+	{
+		data->setInterpretation( IGeomParamTraits<GeomParam>::geometricInterpretation() );
+	};
+
+};
+
+} // namespace
+
+//////////////////////////////////////////////////////////////////////////
+// GeomBaseAlgo implementation
+//////////////////////////////////////////////////////////////////////////
+
+namespace IECoreAlembic
+{
+
+namespace GeomBaseAlgo
+{
+
+void convertUVs( const Alembic::AbcGeom::IV2fGeomParam &uvs, const Alembic::Abc::ISampleSelector &sampleSelector, IECore::Primitive *primitive )
+{
+	if( !uvs.valid() )
+	{
+		return;
+	}
+
+	/// \todo It'd be nice if we stored uvs as a single primitive variable instead of having to split them in two.
+	/// It'd also be nice if we supported indexed data directly.
+	typedef IV2fArrayProperty::sample_ptr_type SamplePtr;
+	SamplePtr sample = uvs.getExpandedValue( sampleSelector ).getVals();
+	size_t size = sample->size();
+
+	FloatVectorDataPtr sData = new FloatVectorData;
+	FloatVectorDataPtr tData = new FloatVectorData;
+	std::vector<float> &s = sData->writable();
+	std::vector<float> &t = tData->writable();
+	s.resize( size );
+	t.resize( size );
+	for( size_t i=0; i<size; ++i )
```

Comment:
I think maybe Andrew and I differ naturally on this one and I don't know that we have something written down as a tiebreaker. In Gaffer at least `for( size_t i = 0; I < size; ++i )` is definitely the prevailing style, but I don't quibble if PRs come in t'other way. In Cortex it seems more split. I suggest we don't worry for now...